config/oidc.go

```diff
@@ -0,0 +1,68 @@
+package config
+
+import "github.com/ngrok/ngrok-go/internal/pb_agent"
+
+type OIDCOption func(cfg *oidcOptions)
+
+type oidcOptions struct {
+	IssuerURL    string
+	ClientID     string
+	ClientSecret string
+	AllowEmails  []string
+	AllowDomains []string
+	Scopes       []string
+}
+
+func (oidc *oidcOptions) toProtoConfig() *pb_agent.MiddlewareConfiguration_OIDC {
+	if oidc == nil {
+		return nil
+	}
+
+	return &pb_agent.MiddlewareConfiguration_OIDC{
+		IssuerURL:    oidc.IssuerURL,
+		ClientID:     oidc.ClientID,
+		ClientSecret: oidc.ClientSecret,
+		AllowEmails:  oidc.AllowEmails,
+		AllowDomains: oidc.AllowDomains,
+		Scopes:       oidc.Scopes,
+	}
+}
+
+// WithOIDC configures this edge with the the given OIDC provider.
+// Overwrites any previously-set OIDC configuration.
+func WithOIDC(issuerURL string, clientID string, clientSecret string, opts ...OIDCOption) HTTPEndpointOption {
```

Comment:
is three required parameters enough to break it out into a struct? probably not, but maybe?

Reply:
Yeah, I think it's *just* few enough to avoid a struct. We've managed to avoid any extra structs in our config interface with the functional options, so I think it makes sense to err on the side of "no structs" unless it's absolutely awful without them.